crypto/aes/aes_ige.c

```diff
@@ -136,7 +136,7 @@ void AES_ige_encrypt(const unsigned char *in, unsigned char *out,
                 in += AES_BLOCK_SIZE;
                 out += AES_BLOCK_SIZE;
             }
-            memcpy(ivec, ivp->data, AES_BLOCK_SIZE);
+            memmove(ivec, ivp->data, AES_BLOCK_SIZE);
```

Comment:
Please note that ivp->data will be pointing at the same address as `in` (or some further blocks in `in`) so there should be no overlap here.

Reply:
This is overlapping when this condition line no: 122 while (len) takes a false branch.

Reply:
but that seems to be impossible if I assume the assertion here will not be failed:
https://github.com/openssl/openssl/blob/d8fa4cf76308924daaf2335c6c0ff2f7334a5b26/crypto/aes/aes_ige.c#L60
together with the precondition that length != 0, therefore the loop must be executed at least once.
since https://github.com/openssl/openssl/blob/d8fa4cf76308924daaf2335c6c0ff2f7334a5b26/crypto/aes/aes_ige.c#L53
and https://github.com/openssl/openssl/blob/d8fa4cf76308924daaf2335c6c0ff2f7334a5b26/crypto/aes/aes_ige.c#L55-L56

Reply:
Exactly. This cannot happen.